lib/rules/no-sync.js

```diff
@@ -27,25 +150,126 @@ module.exports = {
-                    messageId: "noSync",
-                    data: {
-                        propertyName: node.property.name,
-                    },
-                })
+            [ModuleDeclarations](node) {
+                const variables =
+                    context.sourceCode?.getDeclaredVariables?.(node) ??
+                    context.getDeclaredVariables?.(node)
+
+                for (const variable of variables) {
+                    for (const reference of variable.references) {
+                        if (
+                            options?.allowAtRootLevel === true &&
+                            isFunctionScope(reference.from) === false
+                        ) {
+                            continue
+                        }
+
+                        /** @type {import('eslint').Rule.Node} */
+                        const memberExpression = reference.identifier.parent
+                        if (
+                            memberExpression.type !== "MemberExpression" ||
+                            syncRegex.test(memberExpression.property.name) ===
+                                false
+                        ) {
+                            continue
+                        }
+
+                        context.report({
+                            node: memberExpression.property,
+                            messageId: "noSyncMember",
+                            data: {
+                                importName: memberExpression.property.name,
+                            },
+                        })
+                    }
+                }
+            },
+            [AliasDeclarations](node) {
+                const variables =
+                    context.sourceCode?.getDeclaredVariables?.(node) ??
+                    context.getDeclaredVariables?.(node)
+
+                for (const variable of variables) {
+                    for (const reference of variable.references ?? []) {
+                        if (
+                            options?.allowAtRootLevel === true &&
+                            isFunctionScope(reference.from) === false
+                        ) {
+                            continue
+                        }
+
+                        if (node.imported.name === node.local.name) {
+                            context.report({
+                                node: reference.identifier,
+                                messageId: "noSyncMember",
+                                data: { importName: node.imported.name },
+                            })
+                        } else {
+                            context.report({
+                                node: reference.identifier,
+                                messageId: "noSyncAlias",
+                                data: {
+                                    importName: node.imported.name,
+                                    aliasName: variables.at(0)?.name,
+                                },
```

Comment:
out of curiosity: is there a benefit to use `.at()` over `[0]`?
not sure `.at()` was supported since node.js v16.

Reply:
This is just personal preference, I like the way .at behaves for negative numbers and for non numbers

I really didn't think about it that much, from a performance standpoint you probably want to use [0]. But I dont know 🤔

Reply:
Just checked, it was 16.6 - when I am at my pc I will change it to an array index

Reply:
seems its support was added in node.js [v16.8.0](https://node.green/#ES2022-features--at---method-on-the-built-in-indexables), while the package requires node.js `>=16.0.0`.  let's just use `[0]`? 😅